diff -Nur Coro-3.41.orig/Coro/libcoro/coro.c Coro-3.41.chng/Coro/libcoro/coro.c
--- Coro-3.41.orig/Coro/libcoro/coro.c	2006-11-29 12:46:19.000000000 +0100
+++ Coro-3.41.chng/Coro/libcoro/coro.c	2007-02-06 23:01:35.000000000 +0100
@@ -215,8 +215,16 @@
   ctx->env[0].__jmpbuf[0].__pc = (char *)coro_init;
   ctx->env[0].__jmpbuf[0].__sp = (void *)((char *)sptr + ssize);
 #elif defined(__GNU_LIBRARY__) && defined(__amd64__)
+
+#ifndef JB_PC
+#define JB_PC 7
+#define JB_RSP 6
+#define JB_RBP 1
+#endif
+
   ctx->env[0].__jmpbuf[JB_PC]  = (long)coro_init;
   ctx->env[0].__jmpbuf[JB_RSP] = (long)STACK_ADJUST_PTR (sptr,ssize);
+  
 #else
 #error "linux libc or architecture not supported"
 #endif
